src/backend/opencl/magma/magma_blas_clblas.h

```diff
@@ -0,0 +1,74 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <defines.hpp>
+
+#include <clBLAS.h>
+#include <err_clblas.hpp>
+
+// Convert MAGMA constants to clBLAS constants
+clblasOrder          clblas_order_const( magma_order_t order );
+clblasTranspose      clblas_trans_const( magma_trans_t trans );
+clblasUplo           clblas_uplo_const ( magma_uplo_t  uplo  );
+clblasDiag           clblas_diag_const ( magma_diag_t  diag  );
+clblasSide           clblas_side_const ( magma_side_t  side  );
+
+// Error checking
+#define OPENCL_BLAS_CHECK CLBLAS_CHECK
+
+// Transposing
+#define OPENCL_BLAS_TRANS_TYPE clblasTranspose // the type
```

Comment:
I'm also against using `#define`, but I saw quite a lot of it already in the codebase, so I thought it would be OK. But still better to get rid of it :-)

Before I make the changes everywhere let's make sure we understand each other. For example for the OPENCL_BLAS_TRANS_TYPE define I could replace it with (for the CLBlast version):
```
enum class OPENCL_BLAS_TRANS_T {
    NO = static_cast<int>(clblast::Transpose::kNo),
    YES = static_cast<int>(clblast::Transpose::kYes),
    CONJUGATE = static_cast<int>(clblast::Transpose::kConjugate)
};
```
Then in the code we would replace for example `OPENCL_BLAS_CONJ_TRANS` with `OPENCL_BLAS_TRANS_T::CONJUGATE`. We now have a scoped enum that is convertible to the CLBlast version, but that's not going to happen automatically. So there will quite some extra code in the 2 back-ends to handle this conversion. Are there any better ways of handling this? We have to keep in mind that although CLBlast provides a scoped enum (`clblast::Transpose`), clBLAS doesn't since it has a C API.